mutiny-core/src/node.rs

```diff
@@ -569,6 +569,68 @@ impl Node {
         amount_sat: Option<u64>,
         description: String,
         route_hints: Option<Vec<PhantomRouteHints>>,
+    ) -> Result<Invoice, MutinyError> {
+        // the amount to create for the invoice whether or not there is an lsp
+        let (amount_sat, lsp_fee_sat) = if let Some(lsp) = self.lsp_client.clone() {
+            // LSP requires an amount
+            let amount_sat = amount_sat
+                .ok_or(MutinyError::BadAmountError)
+                .and_then(|amount| {
+                    if amount == 0 {
+                        Err(MutinyError::BadAmountError)
+                    } else {
+                        Ok(amount)
+                    }
+                })?;
```

Comment:
can simplify this as `amount_sat.filter(|a| a > 0).ok_or(MutinyError::BadAmountError)`